tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TaskMapper.java

```diff
@@ -69,4 +74,70 @@ private TesExecutor getExecutorFromTesExecutorsList(List<TesExecutor> tesExecuto
                 "executors"));
         return tesExecutors.get(FIRST);
     }
+
+    public TesTask mapToTesTask(PipelineRun run) {
+        final TesTask tesTask = new TesTask();
+        tesTask.setId(String.valueOf(run.getId()));
+        tesTask.setState(createTesState(run.getStatus()));
+        tesTask.setName(run.getPodId());
+        tesTask.setResources(createTesResources(run));
+        tesTask.setExecutors(createListExecutor(run));
+        tesTask.setInputs(createTesInput(ListUtils.emptyIfNull(run.getPipelineRunParameters())));
+        tesTask.setOutputs(createTesOutput(ListUtils.emptyIfNull(run.getPipelineRunParameters())));
+        tesTask.setCreationTime(run.getStartDate().toString());
+        return tesTask;
+    }
+
+    private TesState createTesState(TaskStatus status) {
+        switch (status) {
+            case RUNNING:
+                return TesState.RUNNING;
+            case PAUSED:
+                return TesState.PAUSED;
+            case SUCCESS:
+                return TesState.COMPLETE;
+            case FAILURE:
+                return TesState.EXECUTOR_ERROR;
+            case STOPPED:
+                return TesState.CANCELED;
+            default:
+                return TesState.UNKNOWN;
+        }
+    }
+
+    private List<TesInput> createTesInput(List<PipelineRunParameter> parameters) {
+        final TesInput tesInput = new TesInput();
+        parameters.stream()
+                .filter(pipelineRunParameter -> pipelineRunParameter.getType().contains(INPUT_TYPE))
+                .forEach(pipelineRunParameter -> {
+                    tesInput.setName(pipelineRunParameter.getName());
+                    tesInput.setUrl(pipelineRunParameter.getValue());
+                });
+        return ListUtils.emptyIfNull(Arrays.asList(tesInput));
+    }
+
+    private List<TesOutput> createTesOutput(List<PipelineRunParameter> parameters) {
+        final TesOutput tesOutput = new TesOutput();
+        parameters.stream()
+                .filter(pipelineRunParameter -> pipelineRunParameter.getType().contains(OUTPUT_TYPE))
+                .forEach(pipelineRunParameter -> {
+                    tesOutput.setName(pipelineRunParameter.getName());
+                    tesOutput.setUrl(pipelineRunParameter.getValue());
+                });
+        return ListUtils.emptyIfNull(Arrays.asList(tesOutput));
+    }
+
+    private TesResources createTesResources(PipelineRun run){
+        final TesResources tesResources = new TesResources();
+        tesResources.setPreemptible(run.getInstance().getSpot());
+        tesResources.setDiskGb(new Double(run.getInstance().getNodeDisk()));
+        return tesResources;
+    }
+
+    private List<TesExecutor> createListExecutor(PipelineRun run){
+        final TesExecutor tesExecutor = new TesExecutor();
```

Comment:
lets use for Tes entity annotation @Builder, it this case you will able do somethink like:
```
TesResources.builder()
                      .withPreemptible(run.getInstance().getSpot())
                      .withDiskGb(new Double(run.getInstance().getNodeDisk()));
```